Buildings/Controls/OBC/Utilities/PIDWithAutotuning/FirstOrderAMIGO.mo

```diff
@@ -0,0 +1,649 @@
+Buildings.Controls.OBC.Utilities.PIDWithAutotuning.Relay.Controller</a>.
+</li>
+</ul>
+<p>
+Step 2: Extract parameters of a first-order plus time-delay (FOPTD) model
+</p>
+<ul>
+<li>
+Based on the inputs and outputs from the relay controller, the
+parameters of the FOPTD model is calculated (see
+<a href=\"modelica://Buildings.Controls.OBC.Utilities.PIDWithAutotuning.SystemIdentification\">
+Buildings.Controls.OBC.Utilities.PIDWithAutotuning.SystemIdentification</a>).
+The FOPTD is a simplified representation of the control process.
+</li>
+</ul>
+<p>
+Step 3: Calculate the PID gains
+</p>
+<ul>
+<li>
+The PID gains are calculated with the Approximate M-constrained Integral Gain Optimization (AMIGO) method
+based on the FOPTD model parameters (see
+<a href=\"modelica://Buildings.Controls.OBC.Utilities.PIDWithAutotuning.AutoTuner.AMIGO\">
+Buildings.Controls.OBC.Utilities.PIDWithAutotuning.AutoTuner.AMIGO</a>).
+</li>
+</ul>
+<p>
+This block is implemented using
+<a href=\"modelica://Buildings.Controls.OBC.Utilities.PIDWithInputGains\">
+Buildings.Controls.OBC.Utilities.PIDWithInputGains</a>
+and inherits most of its configuration. However, through the parameter
+<code>controllerType</code>, the controller can only be configured as PI or
+PID controller.
+</p>
+<h4>Autotuning process</h4>
+<p>
+Before the tuning process starts, this block has output from
+<a href=\"modelica://Buildings.Controls.OBC.Utilities.PIDWithInputGains\">
+Buildings.Controls.OBC.Utilities.PIDWithInputGains</a>.
+The PID tuning process starts when a request for performing autotuning occurs,
+i.e., when the value of the boolean input signal <code>triTun</code> changes from
+<code>false</code> to <code>true</code>.
+During the tuning process, the block has the output from a relay controller
+(see <a href=\"modelica://Buildings.Controls.OBC.Utilities.PIDWithAutotuning.Relay.Controller\">
+Buildings.Controls.OBC.Utilities.PIDWithAutotuning.Relay.Controller</a>).
+The PID tuning process ends automatically
+(see details in
+<a href=\"modelica://Buildings.Controls.OBC.Utilities.PIDWithAutotuning.Relay.BaseClasses.TuningMonitor\">
+Buildings.Controls.OBC.Utilities.PIDWithAutotuning.BaseClasses.Relay.TunMonitor</a>),
+at which point this block reverts the output from the PID controller that uses the
+tuned PID parameters.
+</p>
+<p>Note:</p>
+<ul>
+<li>
+If an autotuning is ongoing, i.e., <code>inTunPro.y = true</code>,
+a new request for performing autotuning will be ignored and a warning
+will be generated.
+</li>
+<li>
+If the set point is changed during an autotuning process, a warning will be
+generated. The ongoing tuning process will be halted, and no adjustments will be made
+to the PID parameters.
+</li>
+<li>
+The autotuning must be conducted when the process is in a stable state.
+The user should monitor changes in the independent variables and
+the controlled variables (e.g., mass flow rate, temperature) over time.
```

Comment:
@SenHuang19 : The documentation from here on and below uses "controlled variable", "independent variable" and "measurement". I think you mean by "controlled variable" the "measurement". Can you please use the same term for the same thing. Perhaps easiest is to use "controller input" and "controller output".

Afterwards this is ready to merge.

Reply:
@mwetter: Yes, the "controlled variable" is the same as the "measurement". I'll create a PR to update the document according to your suggestion.